test/csuite/wt5212_incremental_backup/main.c

```diff
@@ -0,0 +1,779 @@
+            break;
+        if (strcmp(filename, last_flist[i].name) == 0) {
+            last_flist[i].exist = true;
+            break;
+        }
+    }
+}
+
+static void
+take_full_backup(WT_SESSION *session, int i)
+{
+    FILELIST *flist;
+    WT_CURSOR *cursor;
+    size_t alloc, count;
+    int j, ret;
+    char buf[1024], f[256], h[256];
+    const char *filename, *hdir;
+
+    /*
+     * First time through we take a full backup into the incremental directories. Otherwise only
+     * into the appropriate full directory.
+     */
+    if (i != 0) {
+        testutil_check(__wt_snprintf(h, sizeof(h), "%s.%d", home_full, i));
+        hdir = h;
+    } else
+        hdir = home_incr;
+    if (i == 0) {
+        testutil_check(__wt_snprintf(
+          buf, sizeof(buf), "incremental=(granularity=1M,enabled=true,this_id=ID%d)", i));
+        testutil_check(session->open_cursor(session, "backup:", NULL, buf, &cursor));
+    } else
+        testutil_check(session->open_cursor(session, "backup:", NULL, NULL, &cursor));
+
+    count = 0;
+    alloc = FLIST_INIT;
+    flist = calloc(alloc, sizeof(FILELIST));
+    testutil_assert(flist != NULL);
+    while ((ret = cursor->next(cursor)) == 0) {
+        testutil_check(cursor->get_key(cursor, &filename));
+        process_file(&flist, &count, &alloc, filename);
+
+        /*
+         * If it is a log file, prepend the path for cp.
+         */
+        if (strncmp(filename, WTLOG, WTLOGLEN) == 0)
+            testutil_check(__wt_snprintf(f, sizeof(f), "%s/%s", logpath, filename));
+        else
+            testutil_check(__wt_snprintf(f, sizeof(f), "%s", filename));
+
+        if (i == 0)
+            /*
+             * Take a full backup into each incremental directory.
+             */
+            for (j = 0; j < ITERATIONS_MULTIPLIER * 3; j++) {
+                testutil_check(__wt_snprintf(h, sizeof(h), "%s.%d", home_incr, j));
+                testutil_check(__wt_snprintf(buf, sizeof(buf), "cp %s/%s %s/%s", home, f, h, f));
+#if 0
+                printf("FULL: Copy: %s\n", buf);
+#endif
+                testutil_check(system(buf));
+            }
+        else {
+#if 0
+            testutil_check(__wt_snprintf(h, sizeof(h), "%s.%d", home_full, i);
+#endif
+            testutil_check(__wt_snprintf(buf, sizeof(buf), "cp %s/%s %s/%s", home, f, hdir, f));
+#if 0
+            printf("FULL %d: Copy: %s\n", i, buf);
+#endif
+            testutil_check(system(buf));
+        }
+    }
+    scan_end_check(ret == WT_NOTFOUND);
+    testutil_check(cursor->close(cursor));
+    finalize_files(flist, count);
+}
+
+static void
+take_incr_backup(WT_SESSION *session, int i)
+{
+    FILELIST *flist;
+    WT_CURSOR *backup_cur, *incr_cur;
+    size_t alloc, count, rdsize, tmp_sz;
+    uint64_t offset, size, type;
+    int j, ret, rfd, wfd;
+    char buf[1024], h[256], *tmp;
+    const char *filename;
+    bool first;
+
+    /*! [incremental backup using block transfer]*/
```

Comment:
This is a documentation label from the example. It should be removed.

Reply:
Fixed